### PR TITLE
Preparing for Release 1.0.0

### PR DESCRIPTION
**Pull request by DevSrSouza**
- Creates and moves ScreenModel API to it own module.
- Remove deprecated functions
- Add binary compatibility validator
- Remove Androidx module and `AndroidScreen`
- Promote Experimental APIs to Stable (removing the annotation)
- Generate the API Dump

### CODE REVIEW

File: samples/android/src/main/java/cafe/adriel/voyager/sample/androidLegacy/LegacyScreenOne.kt [hunk 1]
```diff
@@ -12,12 +12,15 @@ import androidx.compose.runtime.Composable
...
-import cafe.adriel.voyager.androidx.AndroidScreen
+import cafe.adriel.voyager.core.screen.Screen
+import cafe.adriel.voyager.core.screen.ScreenKey
+import cafe.adriel.voyager.core.screen.uniqueScreenKey
 import cafe.adriel.voyager.hilt.getScreenModel
 import cafe.adriel.voyager.navigator.LocalNavigator
 import cafe.adriel.voyager.navigator.currentOrThrow
 
-class LegacyScreenOne : AndroidScreen() {
+class LegacyScreenOne : Screen {
```
Comment (nvkleban) [inline samples/android/src/main/java/cafe/adriel/voyager/sample/androidLegacy/LegacyScreenOne.kt:22]: Why `AndroidScreen` was removed in samples? Will be there any replacement of it and it's ScreenLifecycleProvider implementation?

Comment (nvkleban) [reply samples/android/src/main/java/cafe/adriel/voyager/sample/androidLegacy/LegacyScreenOne.kt:22]: Thank you for the clarification!

**Comment by DevNatan** [comment]
@DevSrSouza can we take a look in TabOptions api design? the current design has "expected" standards for Android like `title` and `icon` but Voyager doesn't have this anywhere else in the API anymore. 

And it's also a little complicated to create these options inside the screen, this should be done by the developer from the outside. Should we see this before the release or can it be done later in a backwards compatible way?

See https://github.com/adrielcafe/voyager/issues/141

**Comment by DevSrSouza** [comment]
> @DevSrSouza can we take a look in TabOptions api design? the current design has "expected standards" for Android like `title` and `icon` but Voyager doesn't have this anywhere else in the API anymore.
> 
> And it's also a little complicated to create these options like `indeX` that we dont have inside the screen, this should be done by the developer from the outside. Should we see this before the release or can it be done later in a backwards compatible way?
> 
> See #141

Yes, we can look into this in next versions and also be backwards compatible.

**Pull request merged**